site/src/Map.jsx

```diff
@@ -1,135 +1,238 @@
-    setPmTilesReady(true)
+    const protocol = new pmtiles.Protocol();
+    maplibregl.addProtocol("pmtiles", protocol.tile);
+    return () => {
+      maplibregl.removeProtocol("pmtiles");
+    };
   }, []);
 
-  const [interactiveLayerIds, setInteractiveLayerIds] = useState(['places']);
-  // const onInteractiveLayersChange = useCallback(layerFilter => {
-  //   setInteractiveLayerIds(MAP_STYLE.layers.map(layer => layer.id).filter(layerFilter));
-  // }, []);
+  const syncInteractiveLayerIds = useCallback(() => {
+    const layers = mapRef.current.getStyle().layers;
+    const layersToShow = layers
+      .filter((layer) => visibleThemes.indexOf(layer.source) >= 0)
+      .map((layer) => layer.id);
+    setInteractiveLayerIds(layersToShow);
+  }, [visibleThemes]);
 
-  const onMouseEnter = useCallback(() => setCursor('pointer'), []);
-  const onMouseLeave = useCallback(() => setCursor('auto'), []);
+  const onMouseEnter = useCallback(() => setCursor("pointer"), []);
+  const onMouseLeave = useCallback(() => setCursor("auto"), []);
 
-  const onClick = useCallback(event => {
+  const onClick = useCallback((event) => {
     const feature = event.features && event.features[0];
-
     if (feature) {
-      setMapEntity(event.features[0].properties);
+      setMapEntity({
+        theme: feature.source,
+        type: feature.sourceLayer,
+        ...feature.properties,
+      });
     } else {
       setMapEntity({});
     }
   }, []);
 
-  function getMapStyle() {
-    let mapStyle;
-
-    if (pmTilesReady){
-     mapStyle =  mode === 'theme-light' ? MAP_STYLE : DARK_MAP_STYLE;
-    } else {
-      mapStyle = undefined;
-    }
-    return mapStyle;
-  }
-
   return (
     <>
       <div className={`map ${mode}`}>
         <MapLibreMap
-          id="myMap"
+          ref={mapRef}
           onMouseEnter={onMouseEnter}
           onMouseLeave={onMouseLeave}
+          onLoad={syncInteractiveLayerIds}
           onClick={onClick}
           cursor={cursor}
           hash={true}
+          mapStyle={MAP_STYLE}
           interactiveLayerIds={interactiveLayerIds}
           initialViewState={INITIAL_VIEW_STATE}
-          mapStyle={getMapStyle()}
-          style={{position: 'fixed', width: '100%', height: 'calc(100vh - 60px)'}}
+          style={{
+            position: "fixed",
+            width: "100%",
+            height: "calc(100vh - 60px)",
+          }}
+          attributionControl={false}
         >
-          <Source id="overture-places" type="vector" url={PLACES_PMTILES_URL}>
-            <Layer {...PLACES_MAP_STYLE} layout={{visibility: interactiveLayerIds.includes('places') ? 'visible' : 'none'}} />
-          </Source>
-          <NavigationControl position='top-right'></NavigationControl>
+          <ThemeSource name="buildings" url={PMTILES_URL} />
+          <ThemeSource name="places" url={PMTILES_URL} />
+          <ThemeSource name="divisions" url={PMTILES_URL} />
+
+          <Layer
```

Comment:
This layer shows the `division` type of the `divisions` theme, and always appears on top so the map is navigable.